tests/jballerina-unit-test/src/test/resources/test-src/runtime/api/identifier_utils/main.bal

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2022, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/test;
+import ballerina/jballerina.java;
+
+public function main() {
+    testIdentifierDecoding();
+    testEscapeSpecialCharacters();
+    testUnescapeSpecialCharacters();
+}
+
+function testIdentifierDecoding() {
+    test:assertEquals(decodeIdentifier("üňĩćőđę_ƈȏɳʂʈ_IL"), "üňĩćőđę_ƈȏɳʂʈ_IL");
+    test:assertEquals(decodeIdentifier("const_IL_123"), "const_IL_123");
+    test:assertEquals(decodeIdentifier(" $0047$0058@$0091`{~_IL"), " /:@[`{~_IL");
+}
+
+function testEscapeSpecialCharacters() {
```

Comment:
Shall we add tests with Unicode codepoints? Eg: `\u{03C0}` etc.

Reply:
Also have a small look at https://github.com/ballerina-platform/ballerina-lang/issues/38334 sample 1. May need to fix and add a negative test.